Realm/RLMBSON.mm

```diff
@@ -324,7 +324,7 @@ Bson RLMConvertRLMBSONToBson(id<RLMBSON> b) {
         case RLMBSONTypeTimestamp:
             return RLMTimestampForNSDate((NSDate *)b);
         case RLMBSONTypeDatetime:
-            return MongoTimestamp(((NSDate *)b).timeIntervalSince1970, 0);
+            return realm::Timestamp(((NSDate *)b).timeIntervalSince1970, 0);
```

Comment:
Something feels a bit off about this– the reason MongoTimestamp was used here is because mongo and realm hold timestamps differently, and they were only compatible with lossiness. 

Can we at least get an integration test demonstrating this? I'm surprised this doesn't break more.